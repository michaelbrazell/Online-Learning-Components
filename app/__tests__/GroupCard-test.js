var React = require('react');
var GroupCard = require('../components/GroupCard.jsx');
var ReactDOM = require('react-dom');
var renderer = require('react-test-renderer');
var {shallow} = require('enzyme');

// Load Sample jsonData
var largeData = 
{"trending_groups":[{"id":2,"name":"Cody Challenge","description":"Problems, submitted by the Cody team and initially hidden, that reveal more problems as you solve them.","problems_count":96,"solved_count":95,"percentage_complete":98.95833333333334,"curated_by":{"id":1,"name":"Cody Team","profile_id":3251402}},{"id":5,"name":"Tiles Challenge","description":"Problems related to the Tiles Programming Contest. See http://www.mathworks.com/matlabcentral/contest/contests/36 ","problems_count":7,"solved_count":2,"percentage_complete":28.57142857142857,"curated_by":{"id":7,"name":"Ned Gulley","profile_id":140947}},{"id":6,"name":"ASEE Challenge","description":"Problems associated with the ASEE conference, June 10-13. Anyone is welcome to play!","problems_count":10,"solved_count":10,"percentage_complete":100.0,"curated_by":{"id":7,"name":"Ned Gulley","profile_id":140947}},{"id":7,"name":"CUP Challenge","description":"Problems developed by MathWorks in collaboration with Cambridge University Press. The problems are suitable for GCSE and A-Level students. ","problems_count":11,"solved_count":11,"percentage_complete":100.0,"curated_by":{"id":7,"name":"Ned Gulley","profile_id":140947}},{"id":8,"name":"Magic Numbers","description":"This is a collection of problems from across Cody with the theme of special \"named\" numbers.","problems_count":19,"solved_count":0,"percentage_complete":0.0,"curated_by":{"id":7,"name":"Ned Gulley","profile_id":140947}},{"id":9,"name":"Modeling & Simulation Challenge","description":"Modeling & Simulation Challenge","problems_count":15,"solved_count":1,"percentage_complete":6.666666666666667,"curated_by":{"id":19664,"name":"John","profile_id":4604828}},{"id":12,"name":"Indexing I","description":"The first collection of indexing problems, selected from problems submitted by members of the MATLAB Central community.","problems_count":27,"solved_count":6,"percentage_complete":22.22222222222222,"curated_by":{"id":26769,"name":"goc3","profile_id":5349647}},{"id":13,"name":"Indexing II","description":"The second collection of indexing problems, selected from problems submitted by members of the MATLAB Central community.","problems_count":22,"solved_count":4,"percentage_complete":18.181818181818183,"curated_by":{"id":26769,"name":"goc3","profile_id":5349647}},{"id":14,"name":"Strings I","description":"The first collection of string-based problems, selected from those submitted by members of the MATLAB Central community.","problems_count":30,"solved_count":2,"percentage_complete":6.666666666666667,"curated_by":{"id":26769,"name":"goc3","profile_id":5349647}}],"progress":[{"id":2,"name":"Cody Challenge","description":"Problems, submitted by the Cody team and initially hidden, that reveal more problems as you solve them.","problems_count":96,"solved_count":95,"percentage_complete":98.95833333333334,"curated_by":{"id":1,"name":"Cody Team","profile_id":3251402}},{"id":5,"name":"Tiles Challenge","description":"Problems related to the Tiles Programming Contest. See http://www.mathworks.com/matlabcentral/contest/contests/36 ","problems_count":7,"solved_count":2,"percentage_complete":28.57142857142857,"curated_by":{"id":7,"name":"Ned Gulley","profile_id":140947}},{"id":19,"name":"Matrix Manipulation I","description":"The first collection of matrix manipulation problems, selected from those submitted by members of the MATLAB Central community.","problems_count":16,"solved_count":4,"percentage_complete":25.0,"curated_by":{"id":26769,"name":"goc3","profile_id":5349647}},{"id":12,"name":"Indexing I","description":"The first collection of indexing problems, selected from problems submitted by members of the MATLAB Central community.","problems_count":27,"solved_count":6,"percentage_complete":22.22222222222222,"curated_by":{"id":26769,"name":"goc3","profile_id":5349647}},{"id":13,"name":"Indexing II","description":"The second collection of indexing problems, selected from problems submitted by members of the MATLAB Central community.","problems_count":22,"solved_count":4,"percentage_complete":18.181818181818183,"curated_by":{"id":26769,"name":"goc3","profile_id":5349647}},{"id":18,"name":"Matrix Patterns I","description":"The first collection of matrix pattern problems, selected from those submitted by members of the MATLAB Central community.","problems_count":18,"solved_count":2,"percentage_complete":11.11111111111111,"curated_by":{"id":26769,"name":"goc3","profile_id":5349647}},{"id":17,"name":"Computational Geometry I","description":"The first collection of computational geometry problems, selected from those submitted by members of the MATLAB Central community.","problems_count":20,"solved_count":2,"percentage_complete":10.0,"curated_by":{"id":26769,"name":"goc3","profile_id":5349647}},{"id":14,"name":"Strings I","description":"The first collection of string-based problems, selected from those submitted by members of the MATLAB Central community.","problems_count":30,"solved_count":2,"percentage_complete":6.666666666666667,"curated_by":{"id":26769,"name":"goc3","profile_id":5349647}},{"id":9,"name":"Modeling & Simulation Challenge","description":"Modeling & Simulation Challenge","problems_count":15,"solved_count":1,"percentage_complete":6.666666666666667,"curated_by":{"id":19664,"name":"John","profile_id":4604828}}],"problems":[{"id":914,"title":"Number of Horns on a unicorn!","created_by":{"id":6052,"name":"Patrick Schuster","score":175,"badges_count":3,"profile_id":2828019,"image_path":"/matlabcentral/fileexchange/images/profilepic_default.gif"},"tags":[{"created_at":"2012-08-16T13:53:23Z","id":864,"name":"unicorn","taggings_count":4,"updated_at":"2012-08-16T13:53:23Z"},{"created_at":"2012-06-05T11:41:09Z","id":708,"name":"fun","taggings_count":121,"updated_at":"2012-06-05T11:41:09Z"},{"created_at":"2014-06-08T09:55:05Z","id":2015,"name":"spam","taggings_count":249,"updated_at":"2014-06-08T09:55:05Z"}],"likes_count":1,"solvers_count":443,"group":{"id":1,"name":"Community"},"solved":false},{"id":42469,"title":"Produce a cosine wave","created_by":{"id":13247,"name":"MathWorks Simulink Team","score":345,"badges_count":5,"profile_id":4054485,"image_path":"/matlabcentral/profiles/4054485.jpg"},"tags":[{"created_at":"2012-07-04T15:42:34Z","id":770,"name":"simulink","taggings_count":39,"updated_at":"2012-07-04T15:42:34Z"},{"created_at":"2012-02-17T22:52:21Z","id":280,"name":"trigonometry","taggings_count":43,"updated_at":"2012-02-17T22:52:21Z"}],"likes_count":2,"solvers_count":441,"group":{"id":9,"name":"Modeling & Simulation Challenge"},"solved":false},{"id":42466,"title":"Add a block to a model","created_by":{"id":13247,"name":"MathWorks Simulink Team","score":345,"badges_count":5,"profile_id":4054485,"image_path":"/matlabcentral/profiles/4054485.jpg"},"tags":[{"created_at":"2012-07-04T15:42:34Z","id":770,"name":"simulink","taggings_count":39,"updated_at":"2012-07-04T15:42:34Z"}],"likes_count":5,"solvers_count":388,"group":{"id":9,"name":"Modeling & Simulation Challenge"},"solved":false},{"id":760,"title":"Duplicate each element of a vector.","created_by":{"id":1309,"name":"Julio","score":1365,"badges_count":7,"profile_id":1018807,"image_path":"/matlabcentral/fileexchange/images/profilepic_default.gif"},"tags":[{"created_at":"2012-01-18T01:00:17Z","id":3,"name":"vectors","taggings_count":103,"updated_at":"2012-01-18T01:00:17Z"}],"likes_count":0,"solvers_count":358,"group":{"id":1,"name":"Community"},"solved":false},{"id":42468,"title":"Produce a sine wave","created_by":{"id":13247,"name":"MathWorks Simulink Team","score":345,"badges_count":5,"profile_id":4054485,"image_path":"/matlabcentral/profiles/4054485.jpg"},"tags":[{"created_at":"2012-07-04T15:42:34Z","id":770,"name":"simulink","taggings_count":39,"updated_at":"2012-07-04T15:42:34Z"},{"created_at":"2012-02-17T22:52:21Z","id":280,"name":"trigonometry","taggings_count":43,"updated_at":"2012-02-17T22:52:21Z"}],"likes_count":3,"solvers_count":339,"group":{"id":9,"name":"Modeling & Simulation Challenge"},"solved":false},{"id":42467,"title":"Add offset to a signal","created_by":{"id":13247,"name":"MathWorks Simulink Team","score":345,"badges_count":5,"profile_id":4054485,"image_path":"/matlabcentral/profiles/4054485.jpg"},"tags":[{"created_at":"2012-07-04T15:42:34Z","id":770,"name":"simulink","taggings_count":39,"updated_at":"2012-07-04T15:42:34Z"}],"likes_count":0,"solvers_count":303,"group":{"id":9,"name":"Modeling & Simulation Challenge"},"solved":false},{"id":824,"title":" Set the array elements whose value is 13 to 0","created_by":{"id":639,"name":"Aurelien Queffurust","score":7795,"badges_count":12,"profile_id":2391181,"image_path":"/matlabcentral/profiles/2391181.jpg"},"tags":[{"created_at":"2012-01-18T01:00:17Z","id":9,"name":"indexing","taggings_count":100,"updated_at":"2012-01-18T01:00:17Z"},{"created_at":"2012-02-09T22:17:06Z","id":242,"name":"logical","taggings_count":87,"updated_at":"2012-02-09T22:17:06Z"},{"created_at":"2012-02-15T15:51:44Z","id":263,"name":"logical indexing","taggings_count":22,"updated_at":"2012-02-15T15:51:44Z"}],"likes_count":0,"solvers_count":300,"group":{"id":12,"name":"Indexing I"},"solved":false},{"id":556,"title":"Stuff the Board","created_by":{"id":7,"name":"Ned Gulley","score":1760,"badges_count":9,"profile_id":140947,"image_path":"/matlabcentral/profiles/140947.jpg"},"tags":[{"created_at":"2012-04-03T17:18:16Z","id":515,"name":"contest","taggings_count":225,"updated_at":"2012-04-03T17:18:16Z"}],"likes_count":6,"solvers_count":295,"group":{"id":5,"name":"Tiles Challenge"},"solved":true},{"id":867,"title":"Replicate elements in vectors","created_by":{"id":5390,"name":"Jean-Marie Sainthillier","score":25380,"badges_count":20,"profile_id":343082,"image_path":"/matlabcentral/profiles/343082.jpg"},"tags":[{"created_at":"2012-01-18T01:00:17Z","id":3,"name":"vectors","taggings_count":103,"updated_at":"2012-01-18T01:00:17Z"},{"created_at":"2012-01-18T01:00:17Z","id":9,"name":"indexing","taggings_count":100,"updated_at":"2012-01-18T01:00:17Z"},{"created_at":"2012-02-19T06:57:51Z","id":288,"name":"basic matlab","taggings_count":348,"updated_at":"2012-02-19T06:57:51Z"}],"likes_count":3,"solvers_count":294,"group":{"id":12,"name":"Indexing I"},"solved":false},{"id":42471,"title":"Make a full wave rectifier","created_by":{"id":13247,"name":"MathWorks Simulink Team","score":345,"badges_count":5,"profile_id":4054485,"image_path":"/matlabcentral/profiles/4054485.jpg"},"tags":[{"created_at":"2012-07-04T15:42:34Z","id":770,"name":"simulink","taggings_count":39,"updated_at":"2012-07-04T15:42:34Z"},{"created_at":"2015-07-29T14:36:01Z","id":13055,"name":"waveforms","taggings_count":2,"updated_at":"2015-07-29T14:36:01Z"},{"created_at":"2015-07-29T14:36:01Z","id":13056,"name":"rectifier","taggings_count":2,"updated_at":"2015-07-29T14:36:01Z"}],"likes_count":1,"solvers_count":277,"group":{"id":9,"name":"Modeling & Simulation Challenge"},"solved":false},{"id":2509,"title":"Wind Chill Computation","created_by":{"id":24594,"name":"Prof. Joe Hummel","score":785,"badges_count":4,"profile_id":4779044,"image_path":"/matlabcentral/profiles/4779044.jpg"},"tags":[{"created_at":"2014-08-14T22:08:31Z","id":2142,"name":"windchill","taggings_count":1,"updated_at":"2014-08-14T22:08:31Z"},{"created_at":"2014-06-25T20:02:30Z","id":2038,"name":"uic","taggings_count":7,"updated_at":"2014-06-25T20:02:30Z"},{"created_at":"2014-06-25T20:02:30Z","id":2039,"name":"cs109","taggings_count":9,"updated_at":"2014-06-25T20:02:30Z"},{"created_at":"2014-08-14T22:08:31Z","id":2143,"name":"straightline computation","taggings_count":4,"updated_at":"2014-08-14T22:08:31Z"}],"likes_count":0,"solvers_count":321,"group":{"id":1,"name":"Community"},"solved":false},{"id":568,"title":"Number of 1s in a binary string","created_by":{"id":2974,"name":"Srivardhini","score":860,"badges_count":3,"profile_id":3162046,"image_path":"/matlabcentral/profiles/3162046.jpg"},"tags":[{"created_at":"2012-02-12T12:29:11Z","id":250,"name":"sum","taggings_count":92,"updated_at":"2012-02-12T12:29:11Z"},{"created_at":"2012-01-26T08:18:49Z","id":46,"name":"binary","taggings_count":66,"updated_at":"2012-01-26T08:18:49Z"}],"likes_count":2,"solvers_count":269,"group":{"id":14,"name":"Strings I"},"solved":false},{"id":42470,"title":"Make a half wave rectifier","created_by":{"id":13247,"name":"MathWorks Simulink Team","score":345,"badges_count":5,"profile_id":4054485,"image_path":"/matlabcentral/profiles/4054485.jpg"},"tags":[{"created_at":"2012-07-04T15:42:34Z","id":770,"name":"simulink","taggings_count":39,"updated_at":"2012-07-04T15:42:34Z"},{"created_at":"2015-07-29T14:36:01Z","id":13055,"name":"waveforms","taggings_count":2,"updated_at":"2015-07-29T14:36:01Z"},{"created_at":"2015-07-29T14:36:01Z","id":13056,"name":"rectifier","taggings_count":2,"updated_at":"2015-07-29T14:36:01Z"}],"likes_count":4,"solvers_count":260,"group":{"id":9,"name":"Modeling & Simulation Challenge"},"solved":false},{"id":2510,"title":"Solving Quadratic Equations (Version 1)","created_by":{"id":24594,"name":"Prof. Joe Hummel","score":785,"badges_count":4,"profile_id":4779044,"image_path":"/matlabcentral/profiles/4779044.jpg"},"tags":[{"created_at":"2014-08-15T08:54:06Z","id":2145,"name":"quadraticequations","taggings_count":2,"updated_at":"2014-08-15T08:54:06Z"},{"created_at":"2014-06-25T20:02:30Z","id":2038,"name":"uic","taggings_count":7,"updated_at":"2014-06-25T20:02:30Z"},{"created_at":"2014-06-25T20:02:30Z","id":2039,"name":"cs109","taggings_count":9,"updated_at":"2014-06-25T20:02:30Z"},{"created_at":"2014-08-14T22:08:31Z","id":2143,"name":"straightline computation","taggings_count":4,"updated_at":"2014-08-14T22:08:31Z"}],"likes_count":1,"solvers_count":279,"group":{"id":1,"name":"Community"},"solved":false}],"player":{"id":17929,"name":"Ashan Peiris","score":3270,"badges_count":5,"profile_id":3296802,"image_path":"/matlabcentral/profiles/3296802.jpg"}}

var completedGroup = 
{"trending_groups":
  [
    {
      "id":7,"name":"CUP Challenge","description":"Problems developed by MathWorks in collaboration with Cambridge University Press. The problems are suitable for GCSE and A-Level students. ","problems_count":11,"solved_count":11,"percentage_complete":100.0,"curated_by":
        {
          "id":7,"name":"Ned Gulley","profile_id":140947
        }
    }
  ]
}

it('Renders with Sample Data', () => {
  const div = document.createElement('div');
  ReactDOM.render(<GroupCard groupsData={completedGroup} />, div);
});

it('Renders with 9 Components', () => {
  var componentCount = largeData.trending_groups.length;
  expect(componentCount).toEqual(9);
});

it('Matches the rendered snapshot', () => {
  const tree = renderer.create(<GroupCard groupsData={completedGroup}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Displays "Complete" when group is complete', () => {
  const tree = renderer.create(<GroupCard groupsData={completedGroup}/>).toJSON();
  expect(tree).toMatchSnapshot();
});