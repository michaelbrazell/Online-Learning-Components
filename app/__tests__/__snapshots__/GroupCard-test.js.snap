// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the rendered snapshot 1`] = `
<div
  className="row add_equal_heights"
>
  <div
    className="col-xs-12 col-sm-6 col-md-4"
  >
    <div
      className="panel panel-default add_text_wrapping group_overview"
      id="group_2"
    >
      <div
        className="panel-body"
      >
        <h4
          className="add_margin_5"
        >
          <a
            href="/matlabcentral/cody/groups/2"
          >
            Cody Challenge
          </a>
        </h4>
        <p
          className="add_margin_40"
        >
          Problems, submitted by the Cody team and initially hidden, that reveal more problems as you solve them.
        </p>
      </div>
      <div
        className="panel-footer"
      >
        <p
          className="pull-left add_font_color_mediumgray add_margin_0"
        >
          95
          /
          96
           Problems
        </p>
        <div
          className="progress add_progress_small add_margin_0"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow={0.9895833333333335}
            className="progress-bar"
            role="progressbar"
            style={
              Object {
                "width": "98.95833333333334%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-xs-12 col-sm-6 col-md-4"
  >
    <div
      className="panel panel-default add_text_wrapping group_overview"
      id="group_5"
    >
      <div
        className="panel-body"
      >
        <h4
          className="add_margin_5"
        >
          <a
            href="/matlabcentral/cody/groups/5"
          >
            Tiles Challenge
          </a>
        </h4>
        <p
          className="add_margin_40"
        >
          Problems related to the Tiles Programming Contest. See http://www.mathworks.com/matlabcentral/contest/contests/36 
        </p>
      </div>
      <div
        className="panel-footer"
      >
        <p
          className="pull-left add_font_color_mediumgray add_margin_0"
        >
          2
          /
          7
           Problems
        </p>
        <div
          className="progress add_progress_small add_margin_0"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow={0.2857142857142857}
            className="progress-bar"
            role="progressbar"
            style={
              Object {
                "width": "28.57142857142857%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-xs-12 col-sm-6 col-md-4"
  >
    <div
      className="panel panel-default add_text_wrapping group_overview"
      id="group_6"
    >
      <div
        className="panel-body"
      >
        <h4
          className="add_margin_5"
        >
          <a
            href="/matlabcentral/cody/groups/6"
          >
            ASEE Challenge
          </a>
        </h4>
        <p
          className="add_margin_40"
        >
          Problems associated with the ASEE conference, June 10-13. Anyone is welcome to play!
        </p>
      </div>
      <div
        className="panel-footer"
      >
        <p
          className="pull-left add_font_color_green add_margin_0"
        >
          Complete
        </p>
        <div
          className="progress add_progress_small add_margin_0"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow={1}
            className="progress-bar add_background_color_green"
            role="progressbar"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-xs-12 col-sm-6 col-md-4"
  >
    <div
      className="panel panel-default add_text_wrapping group_overview"
      id="group_7"
    >
      <div
        className="panel-body"
      >
        <h4
          className="add_margin_5"
        >
          <a
            href="/matlabcentral/cody/groups/7"
          >
            CUP Challenge
          </a>
        </h4>
        <p
          className="add_margin_40"
        >
          Problems developed by MathWorks in collaboration with Cambridge University Press. The problems are suitable for GCSE and A-Level students. 
        </p>
      </div>
      <div
        className="panel-footer"
      >
        <p
          className="pull-left add_font_color_green add_margin_0"
        >
          Complete
        </p>
        <div
          className="progress add_progress_small add_margin_0"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow={1}
            className="progress-bar add_background_color_green"
            role="progressbar"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-xs-12 col-sm-6 col-md-4"
  >
    <div
      className="panel panel-default add_text_wrapping group_overview"
      id="group_8"
    >
      <div
        className="panel-body"
      >
        <h4
          className="add_margin_5"
        >
          <a
            href="/matlabcentral/cody/groups/8"
          >
            Magic Numbers
          </a>
        </h4>
        <p
          className="add_margin_40"
        >
          This is a collection of problems from across Cody with the theme of special "named" numbers.
        </p>
      </div>
      <div
        className="panel-footer"
      >
        <p
          className="pull-left add_font_color_mediumgray add_margin_0"
        >
          19
           Problems
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-xs-12 col-sm-6 col-md-4"
  >
    <div
      className="panel panel-default add_text_wrapping group_overview"
      id="group_9"
    >
      <div
        className="panel-body"
      >
        <h4
          className="add_margin_5"
        >
          <a
            href="/matlabcentral/cody/groups/9"
          >
            Modeling & Simulation Challenge
          </a>
        </h4>
        <p
          className="add_margin_40"
        >
          Modeling & Simulation Challenge
        </p>
      </div>
      <div
        className="panel-footer"
      >
        <p
          className="pull-left add_font_color_mediumgray add_margin_0"
        >
          1
          /
          15
           Problems
        </p>
        <div
          className="progress add_progress_small add_margin_0"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow={0.06666666666666667}
            className="progress-bar"
            role="progressbar"
            style={
              Object {
                "width": "6.666666666666667%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-xs-12 col-sm-6 col-md-4"
  >
    <div
      className="panel panel-default add_text_wrapping group_overview"
      id="group_12"
    >
      <div
        className="panel-body"
      >
        <h4
          className="add_margin_5"
        >
          <a
            href="/matlabcentral/cody/groups/12"
          >
            Indexing I
          </a>
        </h4>
        <p
          className="add_margin_40"
        >
          The first collection of indexing problems, selected from problems submitted by members of the MATLAB Central community.
        </p>
      </div>
      <div
        className="panel-footer"
      >
        <p
          className="pull-left add_font_color_mediumgray add_margin_0"
        >
          6
          /
          27
           Problems
        </p>
        <div
          className="progress add_progress_small add_margin_0"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow={0.2222222222222222}
            className="progress-bar"
            role="progressbar"
            style={
              Object {
                "width": "22.22222222222222%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-xs-12 col-sm-6 col-md-4"
  >
    <div
      className="panel panel-default add_text_wrapping group_overview"
      id="group_13"
    >
      <div
        className="panel-body"
      >
        <h4
          className="add_margin_5"
        >
          <a
            href="/matlabcentral/cody/groups/13"
          >
            Indexing II
          </a>
        </h4>
        <p
          className="add_margin_40"
        >
          The second collection of indexing problems, selected from problems submitted by members of the MATLAB Central community.
        </p>
      </div>
      <div
        className="panel-footer"
      >
        <p
          className="pull-left add_font_color_mediumgray add_margin_0"
        >
          4
          /
          22
           Problems
        </p>
        <div
          className="progress add_progress_small add_margin_0"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow={0.18181818181818182}
            className="progress-bar"
            role="progressbar"
            style={
              Object {
                "width": "18.181818181818183%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-xs-12 col-sm-6 col-md-4"
  >
    <div
      className="panel panel-default add_text_wrapping group_overview"
      id="group_14"
    >
      <div
        className="panel-body"
      >
        <h4
          className="add_margin_5"
        >
          <a
            href="/matlabcentral/cody/groups/14"
          >
            Strings I
          </a>
        </h4>
        <p
          className="add_margin_40"
        >
          The first collection of string-based problems, selected from those submitted by members of the MATLAB Central community.
        </p>
      </div>
      <div
        className="panel-footer"
      >
        <p
          className="pull-left add_font_color_mediumgray add_margin_0"
        >
          2
          /
          30
           Problems
        </p>
        <div
          className="progress add_progress_small add_margin_0"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow={0.06666666666666667}
            className="progress-bar"
            role="progressbar"
            style={
              Object {
                "width": "6.666666666666667%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
